dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguagePluginHandler.java

```diff
@@ -112,29 +114,23 @@ public VersionTypeValidation validateWorkflowSet(Set<SourceFile> sourcefiles, St
             String absolutePath = file.getAbsolutePath();
 
             MinimalLanguageInterface.GenericFileType fileType;
-            switch (file.getType()) {
-            case DOCKSTORE_SMK:
-            case DOCKSTORE_CWL:
-            case DOCKSTORE_WDL:
-            case NEXTFLOW_CONFIG:
-            case NEXTFLOW:
-            case DOCKSTORE_SERVICE_YML:
-            case DOCKSTORE_SERVICE_OTHER:
-            case DOCKSTORE_YML:
+            switch (file.getType().getCategory()) {
+            case GENERIC_DESCRIPTOR:
+            case PRIMARY_DESCRIPTOR:
+            case SECONDARY_DESCRIPTOR:
+            case OTHER:
                 fileType = MinimalLanguageInterface.GenericFileType.IMPORTED_DESCRIPTOR;
                 break;
-            case SMK_TEST_PARAMS:
-            case CWL_TEST_JSON:
-            case WDL_TEST_JSON:
-            case NEXTFLOW_TEST_PARAMS:
-            case DOCKSTORE_SERVICE_TEST_JSON:
+            case TEST_FILE:
                 fileType = MinimalLanguageInterface.GenericFileType.TEST_PARAMETER_FILE;
                 break;
-            default:
+            case CONTAINERFILE:
                 fileType = MinimalLanguageInterface.GenericFileType.CONTAINERFILE;
                 break;
+            default:
+                LOG.error("Could not determine file type category for source file {}", file.getPath());
+                throw new CustomWebApplicationException("Could not determine file type category for source file " + file.getPath(), HttpStatus.SC_METHOD_FAILURE);
```

Comment:
This is a change in behavior; this will "blow up" things, whereas before it would continue.

Reply:
Yeah, this was my proposal. Before it would assume that everything that didn't match was a Dockerfile, which is weird. 
At least this way we should see failures earlier before they hit the DB